quic/s2n-quic-transport/src/ack/pending_ack_ranges.rs

```diff
@@ -0,0 +1,223 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{ack::ack_ranges::AckRanges, interval_set};
+use core::time::Duration;
+use s2n_quic_core::{
+    frame::ack::EcnCounts,
+    packet::number::{PacketNumber, PacketNumberRange},
+};
+
+/// Stores ACK ranges pending processing
+#[derive(Clone, Debug, Default)]
+pub struct PendingAckRanges {
+    ranges: AckRanges,
+    ecn_counts: EcnCounts,
+    ack_delay: Duration,
+}
+
+impl PendingAckRanges {
+    #[inline]
+    pub fn new(ranges: AckRanges, ecn_counts: EcnCounts, ack_delay: Duration) -> Self {
+        PendingAckRanges {
+            ranges,
+            ecn_counts,
+            ack_delay,
+        }
+    }
+
+    /// Extend with a packet number range; dropping smaller values if needed
+    #[inline]
+    pub fn extend<'a>(
+        &mut self,
+        acked_packets: impl Iterator<Item = &'a PacketNumberRange>,
```

Comment:
This needs to return owned ranges for the ACK frame decoder. It can't store those values since they're coming from a varint.

```suggestion
        acked_packets: impl Iterator<Item = PacketNumberRange>,
```